### PR TITLE
Allow setting a jetstream api prefix from a domain

### PR DESCRIPTION
**Pull request by caspervonb**
https://github.com/nats-io/nats-architecture-and-design/issues/52

### CODE REVIEW

File: src/options.rs [hunk 1]
```diff
@@ -524,6 +524,16 @@ impl Options {
         self
     }
 
+    /// Set a custom `JetStream` API prefix from a domain.
+    ///
+    pub fn jetstream_api_prefix_from_domain(mut self, domain: &str) -> Self {
```
Comment (derekcollison) [inline src/options.rs:529]: fn name seems really long, better way to do this?

For Go client its options to getting the JetStream context which I like.

Comment (derekcollison) [reply src/options.rs]: I like it, would just say domain though like we do in Go client. Do APIPrefix() option and Domain() option. In Rust version of course.

Comment (caspervonb) [reply src/options.rs]: As a first pass, let's just embed `JetStreamOptions` into `Options` (this PR).

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**